tests/specs/mcd/verification.k

```diff
@@ -78,19 +78,46 @@ module LEMMAS-MCD-HASKELL [symbolic, kore]
 
     rule chop(Y) => Y +Int pow256 requires #rangeSInt(256, Y) andBool Y <Int 0 [simplification]
 
+    rule maxUInt48 &Int X <=Int maxUInt48 &Int X +Int Y => false
+      requires #rangeUInt(48, X)
+       andBool #rangeUInt(48, Y)
+       andBool notBool #rangeUInt(48, X +Int Y) [simplification]
+
   // ########################
   // Map
   // ########################
 
     rule #range(M:Memory [ K <- V ]                        , START, WIDTH) => #range(M [ K := (V : .ByteArray) ], START, WIDTH)  requires K +Int 1                    ==Int START +Int WIDTH andBool START <=Int K andBool 0 <Int WIDTH andBool 0 <=Int START [concrete(K    , V     ), simplification]
     rule #range(M:Memory [ K <- V ] [ K' := VS':ByteArray ], START, WIDTH) => #range(M [ K := (V : VS')        ], START, WIDTH)  requires K' +Int #sizeByteArray(VS') ==Int START +Int WIDTH andBool START <=Int K andBool 0 <Int WIDTH andBool 0 <=Int START
                                                                                                                                   andBool K +Int 1                    ==Int K'                                                                                [concrete(K, K', V, VS'), simplification]
+  
+  // ########################
+  // Performance Optimizations
+  // ########################
+
+    rule <k> LT maxUInt48 &Int X +Int Y maxUInt48 &Int X => 1 ~> #push ... </k>
+      requires #rangeUInt(48, X)
+       andBool #rangeUInt(48, Y)
+       andBool notBool #rangeUInt(48, X +Int Y) [simplification]
+
+    rule <k> ADD X Y => X +Int Y ~> #push ... </k>
+      requires #rangeUInt(48, X)
+       andBool #rangeUInt(48, Y) [simplification]
```

Comment:
We can't merge rule replacements like this, they are not very safe for updates to the semantics.

Reply:
Instead, add enough functional `runLemma => doneLemma` style claims that each of the new lemmas you add is tested, and then we don't have to enable this test on CI until th ebackend issues that prompted these additions are resolved.

Reply:
I used these rules to workaround a performance bottleneck in the Haskell backend. Now that we have [fixed the original performance issue](https://github.com/kframework/kore/pull/2800) directly in the backend, these rules are no longer needed.

With the workaround, the test needed roughly 6m to succeed on my machine. With the workaround removed and the proper fix applied, performance is even better at roughly 3~4m.